### PR TITLE
[#103010668] normalize and update bosh manifests

### PR DESCRIPTION
**Pull request by mtekel**
https://www.pivotaltracker.com/n/projects/1275640/stories/103010668
Depends on https://github.com/alphagov/cf-terraform/pull/24

Because of maintenance and debug, we want to have as similar microbosh manifests as possible between GCE and AWS. To achieve that we take same approach as with CF manifest unification. We define a common template for both and cloud provider specific stubs. We use spiff to generate the final manifest. Terraform is no longer involved in modifying/rendering bosh manifest.

We also:
- disable ressurector on GCE (see [#103979660](https://www.pivotaltracker.com/n/projects/1275640/stories/103979660))
- update bosh release to v207 for both
- update bosh stemcell to 3074 for both
- update CF stemcell to 3074 for both
- enable smoketests for AWS
- deploy DEAs to 2 zones (just one zone with 2 DEAs previously)
# Testing

Deploy AWS and GCE environments from scratch. Run smoketests to confirm functionality. You can also test applying against existing environment. The update should run cleanly.
# Reviewing

Not @mtekel or @keymon

### CODE REVIEW

File: aws/variables.tf [hunk 1]
```diff
@@ -77,3 +77,14 @@ variable "dns_zone_name" {
...
+
+# Terraform currently only has limited support for reading environment variables
+# Variables for use with terraform must be prefexed with 'TF_VAR_'
+# These two variables are passed in as environment variables named:
+# TF_VAR_AWS_ACCESS_KEY_ID and TF_VAR_AWS_SECRET_ACCESS_KEY respectively
+variable "AWS_ACCESS_KEY_ID" {
+  description = "AWS access key to be pass to the bosh CPI"
+}
+variable "AWS_SECRET_ACCESS_KEY" {}
+  description = "AWS secret access key to be pass to the bosh CPI"
```
Comment (mtekel) [inline aws/variables.tf]: {} looks like a typo... breaks TF

Comment (keymon) [inline aws/variables.tf]: You are right, sorry. Fixed.

**Comment by keymon** [comment]
I rebased on master to get #26

**Comment by keymon** [comment]
(sorry, I click the close button by accident)

**Comment by dcarley** [comment]
Some thoughts on the commits while I wait for the builds to run. Sorry, can't comment on the diff view, because these don't necessarily relate to single lines or commits..
- Can commits d1e0935d1be6261b1c2346ec293f830e31efaa74 and 6496a5d219ea4ea2505e133a14b46c73d45a253c be squashed together? I didn't understand how the first commit was being used or would work until I got the to the second commit.
- Could you add some detail to the commits which update the stemcell and release versions to explain why we're doing that. Is it just for consistency, or to fix the SSH issue on GCE, or both/other?
- How does badc67c1d5208445e4a0ee3acab5a76e2fea6d5c differ from the other commits that change stemcell versions? Should this be squashed into the earlier two commits for AWS and GCE?
- Can we remove these entries from `.gitignore` now that we're not rendering templates on the local machine?

```
aws/manifest.yml
gce/manifest.yml
aws/cf-manifest.yml
gce/cf-manifest.yml
```

**Comment by keymon** [comment]
Yes, commit d1e0935 is wrong because we move it to a different PR, #24, and we rebased this PR, but we did not update the message. sorry for that. I will fix it.

About how  badc67c differs from the rests. The other ones refer to the bosh manifest, meanwhile the other to provision.sh which is used by cloudfoundry. 

I will update the commits to add a better description.

**Comment by mtekel** [comment]
Updating stemcell versions enables us to run the latest BOSH and CF releases. Running latest versions of CF/BOSH is important for our evaluation - we want to test what we might run in production. Unless we want to stick to some older version forever. In case of updating GCE stemcell, it also fixes ssh issue. Having same versions of everything is the point of this story - to minimize differences between our deployments.

**Comment by dcarley** [comment]
That's a good description of _why_ and your thought process. Please make sure you put it into the commit messages though. The value is that you won't need to explain that to each person (like me) that asks and hypothetically, if we realise in a weeks time that the version updates have broken something else and you happen to be on holiday, then we as a team can make an informed decision about whether we can roll it forwards or backwards.

Not sure I strictly agree about us testing the latest and greatest version. It may cause more disruption than good. It's worth noting that we have a [story to test upgrades](https://www.pivotaltracker.com/story/show/103902980). That's by-the-by though.

**Comment by keymon** [comment]
Anyway, I do agree we should upgrade to the latest bosh release to be able to run the latest stemcell (with the SSH patch). About the story to test upgrades, that is more about test the upgrade of CF rather than bosh (although can be both, sure)

**Comment by keymon** [comment]
@dcarley I rebased this to reflect your latest comments.

**Comment by dcarley** [comment]
Provisioned new environments on AWS and GCE. Tests ran fine. Commits look good too. Thanks :+1:

**Pull request merged**